workflow/rules/bwa_mem.smk

```diff
@@ -11,10 +11,17 @@ rule bwa_mem:
         bam=temp("alignment/bwa_mem/{sample}_{type}.bam"),
     params:
         index=config["reference"]["fasta"],
-        extra=config["bwa_mem"]["extra"],
-        sorting=config["bwa_mem"].get("sort", "samtools"),
-        sort_order=config["bwa_mem"].get("sort_order", "coordinate"),
-        sort_extra="-@ %s" % str(config["bwa_mem"]["threads"]),
+        extra="%s %s"
+        % (
+            config.get("bwa_mem", {}).get("extra", ""),
```

Comment:
For me, it is a bit confusing to have a field called `extra` as part of the `config.yaml` which is adapted from the wrapper and then adding another field called `read_group` which is a specific `extra` in addition. I would either keep `extra` and include everything that is handled by the wrapper via `extra` or split it up for the different "extra" things that one might want to include.

Reply:
I wanted to do this change since:
1. It feels strange to have extra as a required parameter 
2. We want to have read group information, but it should be changeable
3. If we set extra in the config I don't want to accidentally remove read group information, right now you need to remember to set it

I could make a code change to the wrapper with a read_group parameter, but this was a faster solution.

Reply:
I understand. I just feel the more we deviate from what we agreed on, the more confusion it will be. If everything follows the standard and then suddenly we need to remember a minor detail, and I feel there are more and more deviations, it becomes more and more difficult to use those modules. But that is just my opinion.

Reply:
How does adding a read_group settings for bwa_mem, deviate from what we have written in our doc? 

With the current implementation you need to remember to add read group information? And if you only add something else via extra the "read group" will not be added.

Reply:
As far as I remember, we discussed that each field in the `params` section should be reflected in `config.yaml`. But I remember probably wrong. Do it the way you think is best.

Reply:
does that mean that you would like it to look like

```
params:
        index=config["reference"]["fasta"],
        extra= config.get("bwa_mem", {}).get("extra", ""),
        read_group=config.get("bwa_mem", {}).get("read_group", "-R '@RG\\tID:{sample}\\tSM:{sample}\\tPL:illumina\\tPU:{sample}' -v 1 "),
        sorting=config.get("bwa_mem", {}).get("sort", "samtools"),
        sort_order=config.get("bwa_mem", {}).get("sort_order", "coordinate"),
        sort_extra="-@ %s"
        % str(config.get("bwa_mem", config["default_resources"]).get("threads", config["default_resources"]["threads"])),
```

Reply:
But if it's not possible? The wrapper would need to be modified to allow for a read_group parameter.

Reply:
As I said, do it as you think is best.